client/github_client_manager.go

```diff
@@ -0,0 +1,115 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/bradleyfalzon/ghinstallation/v2"
+	"github.com/google/go-github/v45/github"
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/mattermost/release-bot/version"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	ClientCacheSize = 64
+)
+
+type GithubClientManager interface {
+	Get(installationID int64) (*github.Client, error)
+	CreateToken(installationID int64) (AccessToken, error)
+}
+
+type AccessToken interface {
+	IsExpired() bool
+	GetToken() string
+}
+
+type clientCache struct {
+	appID              int64
+	privateKeyFile     string
+	userAgent          string
+	cache              *lru.Cache
+	transport          http.RoundTripper
+	appClient          *github.Client
+	installationTokens map[int64]AccessToken
+}
+
+type accessToken struct {
+	token     string
+	expiresAt time.Time
+}
+
+func NewManager(appID int64, privateKeyFile string) (GithubClientManager, error) {
+	cache, err := lru.New(ClientCacheSize)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create cache")
+	}
+	version := version.Full()
+	itr, err := ghinstallation.NewAppsTransportKeyFromFile(http.DefaultTransport, appID, privateKeyFile)
```

Comment:
same here, everything related to `New*` it's a good practice to be a dependency in the factory method so you can mock easier.